### PR TITLE
3084 new module svaba

### PR DESCRIPTION
**Pull request by kubranarci**
## PR checklist

Closes #XXX 

- [ done ] This comment contains a description of changes (with reason).
- [ done ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ done ] Remove all TODO statements.
- [ done ] Emit the `versions.yml` file.
- [ done ] Follow the naming conventions.
- [ done ] Follow the parameters requirements.
- [ done ] Follow the input/output options guidelines.
- [ done ] Add a resource `label`
- [ done ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ done ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by kubranarci** [comment]
pytest workflow is giving an error since I had to edit BWA_INDEX to get the proper output for this module. the test fails here since BWA_INDEX is original here.

**Comment by nvnieuwk** [comment]
I've updated the branch, that should bring in the changes from the repository :)

File: modules/nf-core/svaba/main.nf
Comment (nvnieuwk) [inline modules/nf-core/svaba/main.nf:50]: Does Svaba also have long forms of these flags (like `--genome` for fasta for example?)

File: modules/nf-core/svaba/meta.yml
Comment (nvnieuwk) [inline modules/nf-core/svaba/meta.yml:18]: Can you also add the other meta fields here?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/svaba/main.nf
Comment (kubranarci) [reply modules/nf-core/svaba/main.nf:50]: there is

File: modules/nf-core/svaba/main.nf
Comment (nvnieuwk) [reply modules/nf-core/svaba/main.nf:24]: Yes indeed! bgzipped output is preferred :)

File: modules/nf-core/svaba/main.nf
Comment (nvnieuwk) [reply modules/nf-core/svaba/main.nf:50]: Can you use these? It's more clear to future users what these flags do

File: modules/nf-core/svaba/meta.yml
Comment (kubranarci) [reply modules/nf-core/svaba/meta.yml:18]: done

File: modules/nf-core/svaba/main.nf
Comment (kubranarci) [reply modules/nf-core/svaba/main.nf:68]: done

**Comment by kubranarci** [comment]
problem is the same: pytest workflow is giving an error since I had to edit BWA_INDEX to get the proper output for this module. the test fails here since BWA_INDEX is original here.

Tests were ok in my local. Can you have one more look at this? @nvnieuwk

File: modules/nf-core/svaba/main.nf
Comment (kubranarci) [reply modules/nf-core/svaba/main.nf:21]: I guess it would make more sense if we left this choice to the user itself, no?

Comment (nvnieuwk) [reply modules/nf-core/svaba/main.nf:21]: The most flexible option for the user is to supply an output channel for every file. See [here](https://nf-co.re/docs/contributing/modules#inputoutput-options) part 6

Comment (kubranarci) [reply modules/nf-core/svaba/main.nf:21]: I dont like it but I will do it :D

**Comment by nvnieuwk** [comment]
> problem is the same: pytest workflow is giving an error since I had to edit BWA_INDEX to get the proper output for this module. the test fails here since BWA_INDEX is original here.
> 
> Tests were ok in my local. Can you have one more look at this? @nvnieuwk

It's possible that your reference files get overwritten by the reference in the BWA index. Maybe you can tackle this in another way? Is there an option to supply the BWA index separately?

File: modules/nf-core/svaba/main.nf
Comment (nvnieuwk) [reply modules/nf-core/svaba/main.nf:21]: Yeah sorry 😁, but we value the flexibility this approach gives

**Comment by kubranarci** [comment]
> > problem is the same: pytest workflow is giving an error since I had to edit BWA_INDEX to get the proper output for this module. the test fails here since BWA_INDEX is original here.
> > Tests were ok in my local. Can you have one more look at this? @nvnieuwk
> 
> It's possible that your reference files get overwritten by the reference in the BWA index. Maybe you can tackle this in another way? Is there an option to supply the BWA index separately?

one can provide a directory including BWA indexes, which is I do here. Otherwise indexes could be given separately as well. But I don't know how to generate those in test case. The problem here is, they coded BWA_INDEX output as genome.indexes but svaba requires as them gemome.fasta.indexes. I corrected it in my local for the sake of tests. But I did not push it for the sake of others :D

**Comment by kubranarci** [comment]
> Aaah okay I get it, No problem then! Can you change the tests to stubs (by adding `-stub` to the command in the `test.yml`? This way tests will pass here and will have to be rerun locally everytime someone updates the module

Can you do it?  I am not sure how to do

File: tests/modules/nf-core/svaba/test.yml
Comment (nvnieuwk) [inline tests/modules/nf-core/svaba/test.yml:31]:
```suggestion
  command: nextflow run ./tests/modules/nf-core/svaba -entry test_svaba -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/svaba/nextflow.config -stub
```

File: modules/nf-core/svaba/main.nf
Comment (nvnieuwk) [inline modules/nf-core/svaba/main.nf:76]:
```suggestion
        svaba: \$(echo \$(svaba --version 2>&1) | sed 's/[^0-9.]*\\([0-9.]*\\).*/\\1/' )
    END_VERSIONS
    """
        stub:
    """
    touch ${prefix}.vcf.gz
    touch ${prefix}.bps.txt.gz
    touch ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        svaba: \$(echo \$(svaba --version 2>&1) | sed 's/[^0-9.]*\\([0-9.]*\\).*/\\1/' )
```

I think this should fix the versions.yml not being created correctly

**Comment by kubranarci** [comment]
It stuck on queue forever

**Comment by kubranarci** [comment]
This should not be failing any more :/

**Comment by nvnieuwk** [comment]
> I don't know why bu VSCODE stopped synchronizing my push request on github :/ I had to copy changes manually and so made some mistakes..

Could be because of the massive traffic in this repository during the hackathon (I had that a couple of times too)

**Comment by kubranarci** [comment]
md5 check fails for stub :/ I started to hate this. I couldn't run with stub in my local

**Comment by nvnieuwk** [comment]
> md5 check fails for stub :/ I started to hate this. I couldn't run with stub in my local

That's completely normal because stubs create empty files. You can remove the md5 checks when running stubs

**Comment by kubranarci** [comment]
> > md5 check fails for stub :/ I started to hate this. I couldn't run with stub in my local
> 
> That's completely normal because stubs create empty files. You can remove the md5 checks when running stubs

finally ready :D 👯‍♀️

**Comment by nvnieuwk** [comment]
The honor is yours to press the merge button 🥳

**Comment by kubranarci** [comment]
> The honor is yours to press the merge button 🥳

I think It requires your approval first :D

**Review by nvnieuwk — APPROVED** [review]
You're right, I thought I already approved :p

**Pull request merged**
